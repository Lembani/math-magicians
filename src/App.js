import React from 'react';

import Calculator from './components/Calculator';

class App extends React.PureComponent {
  render() {
    return (
      <div className="app">
        <Calculator />
      </div>
    );
  }
}

export default App;
