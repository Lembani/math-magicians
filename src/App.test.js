import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from './App';

describe('If app renders correctly', () => {
  test('App is in document', () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const navElement = document.querySelector('.navbar');
    expect(navElement).toBeInTheDocument();
  });

  it('Renders to the screen', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
