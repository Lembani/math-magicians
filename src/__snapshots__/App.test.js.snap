// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If app renders correctly Renders to the screen 1`] = `
<div
  className="app"
>
  <nav
    className="navbar"
  >
    <a
      className="brand nav-link"
      href="/"
      onClick={[Function]}
    >
      Math Magicians
    </a>
    <ul
      className="nav-list-container"
    >
      <li
        className="nav-list-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      |
      <li
        className="nav-list-item"
      >
        <a
          className="nav-link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      |
      <li
        className="nav-list-item"
      >
        <a
          className="nav-link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <h2
    className="home-welcome"
  >
    Welcome to our page!
  </h2>
  <p
    className="para-one"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Ullam magnam, magni autem cupiditate consequatur tempore temporibus quia ab perferendis laudantium eos voluptates sint, saepe ipsum reprehenderit impedit consectetur at veniam architecto, culpa alias voluptatum itaque? Sint aperiam, beatae explicabo quis totam amet. Nihil praesentium, velit exercitationem animi, ipsum quibusdam eveniet distinctio nobis repudiandae laudantium sunt dolorum aliquam? Eius, aliquam! Architecto maiores explicabo facilis? Ullam repellat cumque delectus incidunt? Nam quam tempora doloribus sit, velit corrupti natus quis mollitia exercitationem dicta amet nihil molestias consequuntur vel consectetur vero eaque quas.
  </p>
  <p
    className="para-two"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Ullam magnam, magni autem cupiditate consequatur tempore temporibus quia ab perferendis laudantium eos voluptates sint, saepe ipsum reprehenderit impedit consectetur at veniam architecto, culpa alias voluptatum itaque? Sint aperiam, beatae explicabo quis totam amet. Nihil praesentium, velit exercitationem animi, ipsum quibusdam eveniet distinctio nobis repudiandae laudantium sunt dolorum aliquam? Eius, aliquam! Architecto maiores explicabo facilis? Ullam repellat cumque delectus incidunt? Nam quam tempora doloribus sit, velit corrupti natus quis mollitia exercitationem dicta amet nihil molestias consequuntur vel consectetur vero eaque quas.
  </p>
</div>
`;
